### ISSUE TITLE
wq: resource summary files not cleaned up on early termination

### ISSUE THREAD

**Issue by btovar** [cooperative-computing-lab/cctools#2723]
When using `q.enable_monitoring()`, if the queue is terminated after some tasks have been retrieved, but before they have returned to via `q.wait()`, the temporary summary files are left in the working directory.

We can't do much if the queue is terminated with a sigkill, but it would be cleaner to put all these files in a single temporary directory.

**Comment by dthain** [comment]
Is this really specific to resource summary files, or to output files of the task in general?

**Comment by btovar** [comment]
Only resource summary files, as those are output files that CCTools adds to the task. We never cleanup output files declared by the user.

**Comment by dthain** [comment]
I think I'm ok with this behavior.  There are a variety of cleanup issues that need to be handled if a queue is prematurely terminated, and I think the right way to go is to run the whole application in a subdirectory, not just some of its files.

**Comment by btovar** [comment]
One thing I don't like is that the summary files is our mess, not the user's mess. A user may not know whether it is safe to rm the files, and when it is safe to do so. Further, they may be tempted to do a 'rm wq-*' to get rid of them, and that's just asking for trouble.

**Comment by dthain** [comment]
Well, this points to a need to have "transparent" data returned from a task.  It might be interesting to pull back an output file into a memory buffer, and then just hang that off the task until requested.  The opposite of an input buffer...

**Comment by btovar** [comment]
Yes, I like this idea. The resource summaries know how to parse themselves from a string already.

**Comment by btovar** [comment]
Closing as wq is now in maintenance mode.
